samplers/samplers.go

```diff
@@ -317,16 +317,19 @@ func (h *Histo) Flush(interval time.Duration, percentiles []float64, aggregates
 			Tags:       tags,
 			MetricType: "gauge",
 		})
-		if (aggregates.Value&AggregateAverage) == AggregateAverage && h.LocalWeight != 0 {
-			// we need both a rate and a non-zero sum before it will make sense
-			// to submit an average
-			metrics = append(metrics, DDMetric{
-				Name:       fmt.Sprintf("%s.avg", h.Name),
-				Value:      [1][2]float64{{now, h.LocalSum / h.LocalWeight}},
-				Tags:       tags,
-				MetricType: "gauge",
-			})
-		}
+	}
+
+	if (aggregates.Value&AggregateAverage) == AggregateAverage && h.LocalSum != 0 && h.LocalWeight != 0 {
+		// we need both a rate and a non-zero sum before it will make sense
+		// to submit an average
+		tags := make([]string, len(h.Tags))
```

Comment:
This isn't related, but in reviewing this I noticed that we make a new `tags` for every possible emission here (average, sum, each percentile, etc). Would you mind taking the allocation of `tags` and moving it up to around where we allocate `metrics`? Then each if the `if` blocks can just use `tags` and we save allocating a bunch!